src/phoneField/index.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { PureComponent } from 'react'
+import cc from 'classcat'
+import isEmpty from 'lodash.isempty'
+import isEqual from 'lodash.isequal'
+import SelectField from 'selectField'
+import TextField from 'textField'
+import countryTelData from 'country-telephone-data'
+import style from 'phoneField/style'
+
+
+interface PhoneFieldOnChangeParameters {
+  name: string
+  value: object
+}
+interface PhoneFieldProps  {
+  readonly id?: string
+  readonly name: string
+  readonly className?: string
+  readonly defaultPhoneCountryCodeValue?: string
+  readonly phoneCountryCodeLabel?: string
+  readonly phoneNumberLabel?: string
+  readonly phoneNumberPlaceholder?: string
+  readonly defaultphoneNumberValue?: string
+  readonly countriesWhiteList?: string[]
+  readonly customCountriesName?: customCountriesName
+  readonly isInline?: boolean
+  readonly onChange?: (obj: PhoneFieldOnChangeParameters) => void
+}
+
+interface PhoneFieldState {
+  countriesDatas: object[]
+  countriesWhiteList: string[]
+  phoneNumber: string
+  phoneCountryCode: string
+  completePhoneNumber: string
+  // Make the type of [key: string]: string etc... assignable for setState()
+  [key: string]: string | string[] | object[] | number | boolean
+}
+
+const filterIso2 = (countriesList: string[]) => countryTelData.allCountries.filter(
+  country => countriesList.includes(country.iso2)
+)
+
+const findIso2 = (countryDefault: string) => countryTelData.allCountries.find(
+  country => country.iso2 === countryDefault
+)
```

Comment:
Add a default value here, to ensure the usage of an object

Reply:
Done